src/sentry/sentry_metrics/indexer/redis_mock.py

```diff
@@ -0,0 +1,150 @@
+import hashlib
+from typing import Any, Dict, List, Optional, Union
+
+from django.conf import settings
+
+from sentry.utils.redis import redis_clusters
+
+from .base import StringIndexer
+
+
+def get_client() -> Any:
+    return redis_clusters.get(settings.SENTRY_METRICS_INDEXER_REDIS_CLUSTER)
+
+
+def get_int(string: str) -> int:
+    return int.from_bytes(hashlib.md5(string.encode("utf-8")).digest(), "big") % (10 ** 8)
+
+
+class RedisMockIndexer(StringIndexer):
+    """
+    Temporary mock string indexer that uses Redis to store data.
+    """
+
+    def _get_key(self, org_id: str, instance: Union[str, int]) -> str:
+        if isinstance(instance, str):
+            return f"temp-metrics-indexer:{org_id}:1:str:{instance}"
+        elif isinstance(instance, int):
+            return f"temp-metrics-indexer:{org_id}:1:int:{instance}"
+        else:
+            raise Exception("Invalid: must be string or int")
+
+    def _bulk_record(self, org_id: str, unmapped: Dict[str, None]) -> Dict[str, int]:
+        """
+        Take a mapping of strings {"metric_id`": None} and populate the ints
+        for the corresponding strings.
+        This includes save the following key value pairs in Redis
+            "temp-metrics-indexer:{org_id}:1:str:{instance}" -> int
+            "temp-metrics-indexer:{org_id}:1:int:{instance}" -> string
+        """
+
+        redis_key_values: Dict[str, Union[str, int]] = {}
+        mapped_ints: Dict[str, int] = {}
+
+        for string in unmapped.keys():
+            # use hashlib instead of hash() because the latter uses a random value (unless PYTHONHASHSEED
+            # is set to an integer) to seed hashes of strs and bytes
+            # https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHASHSEED
+            int_value = get_int(string)
+            mapped_ints[string] = int_value
+
+            int_key = self._get_key(org_id, int_value)
+            string_key = self._get_key(org_id, string)
+
+            redis_key_values[string_key] = int_value
+            redis_key_values[int_key] = string
+
+        get_client().mset(redis_key_values)
```

Comment:
Please set an expiry time. If this is used in the production environment while we develop, we do not want these keys to be lingering around.